services/worker/src/worker/job_runners/split/duckdb_index.py

```diff
@@ -0,0 +1,263 @@
+    split_names_best_response = get_previous_step_or_raise(
+        kinds=["config-split-names-from-streaming", "config-split-names-from-info"], dataset=dataset, config=config
+    )
+    try:
+        splits_content = split_names_best_response.response["content"]["splits"]
+    except Exception as e:
+        raise PreviousStepFormatError("Previous step did not return the expected content.", e) from e
+
+    if split not in [split_item["split"] for split_item in splits_content]:
+        raise SplitNotFoundError(f"The split '{split}' does not exist for the config '{config}' of the dataset.")
+
+    # get parquet urls and dataset_info
+    config_parquet_and_info_step = "config-parquet-and-info"
+    parquet_and_info_best_response = get_previous_step_or_raise(
+        kinds=[config_parquet_and_info_step],
+        dataset=dataset,
+        config=config,
+    )
+    content_parquet_and_info = parquet_and_info_best_response.response["content"]
+    try:
+        split_parquet_files = [
+            parquet_file
+            for parquet_file in content_parquet_and_info["parquet_files"]
+            if parquet_file["config"] == config and parquet_file["split"] == split
+        ]
+
+        split_parquets_size = sum(parquet_file["size"] for parquet_file in split_parquet_files)
+
+        if split_parquets_size > max_parquet_size_bytes:
+            raise SplitWithTooBigParquetError(
+                f"The indexing is limited to split parquets under {max_parquet_size_bytes} bytes. "
+                f"Current size of sum of split parquets is {split_parquets_size} bytes."
+            )
+
+        parquet_urls = [parquet_file["url"] for parquet_file in split_parquet_files]
+
+        if not parquet_urls:
+            raise ParquetResponseEmptyError("No parquet files found.")
+
+        # get the features
+        features = content_parquet_and_info["dataset_info"]["features"]
+        column_names = ",".join(list(features.keys()))
+
+        # look for string columns
+        string_columns = [
+            column
+            for column, feature in features.items()
+            if "dtype" in feature
+            and "_type" in feature
+            and feature["dtype"] == STRING_FEATURE_DTYPE
+            and feature["_type"] == VALUE_FEATURE_TYPE
+        ]
+        if not string_columns:
+            raise NoIndexableColumnsError("No string columns available to index.")
+
+    except KeyError as e:
+        raise PreviousStepFormatError(
+            f"Previous step '{config_parquet_and_info_step}' did not return the expected content.", e
+        ) from e
+
+    # configure duckdb extensions
+    duckdb.execute(INSTALL_EXTENSION_COMMAND.format(extension="httpfs"))
+    duckdb.execute(LOAD_EXTENSION_COMMAND.format(extension="httpfs"))
+    duckdb.execute(INSTALL_EXTENSION_COMMAND.format(extension="fts"))
+    duckdb.execute(LOAD_EXTENSION_COMMAND.format(extension="fts"))
+
+    # index all columns
+    if duckdb_index_file_directory is None:
```

Comment:
we could instead force `duckdb_index_file_directory` to be str, not Optional[str], no?